app/controllers/api_controller/parser.rb

```diff
@@ -2,6 +2,11 @@ class ApiController
   module Parser
     def parse_api_request
       @req = RequestAdapter.new(request, params)
+
+      if !collection_option?(:arbitrary_resource_path) && @req.collection_path_elements.length > 4
+        raise BadRequestError,
```

Comment:
yes intentional.  400 BadRequest/malformed URL in this case as per our specification.